hackathon_site/event/test_api.py

```diff
@@ -193,3 +194,84 @@ def test_leave_with_not_empty_team(self):
         self.assertEqual(
             Team.objects.filter(team_code=old_team.team_code).exists(), True
         )
+
+
+class EventTeamListsViewTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.team = Team.objects.create()
+        self.team2 = Team.objects.create()
+        self.team3 = Team.objects.create()
+
+        self.view = reverse("api:event:team-list")
+
+    def _login(self):
+        self.client.login(username=self.user.username, password=self.password)
+
+    def _login_permission(self):
+        self.client.login(username=self.user.username, password=self.password)
+        self.user.user_permissions.add(
+            Permission.objects.get(codename="view_team", pk="28")
+        )
+
+    def _build_filter_url(self, **kwargs):
+        return (
+            self.view + "?" + "&".join([f"{key}={val}" for key, val in kwargs.items()])
+        )
+
+    def test_team_get_no_permission(self):
```

Comment:
Still need a test for not logged in as well, which should have a status of HTTP 401 Unauthorized